lattices/src/algebra.rs

```diff
@@ -348,6 +348,57 @@ pub fn idempotency<S: Debug + PartialEq + Clone, const N: usize>(
     Ok(())
 }
 
+/// Defines the linearity property
+/// q is linear with respect to some group operation + if q(a+b) = q(a) + q(b)
+/// This is the same as q being a group homomorphism
+// As defined in the paper "DBSP: Automatic Incremental View Maintenance for Rich Query Languages"
+pub fn linearity<S: Debug + PartialEq + Clone, R: Debug + PartialEq + Clone>(
+    items: &[S],
+    f: impl Fn(S, S) -> S, // The base operation of the algebraic structure for state
+    g: impl Fn(R, R) -> R, // The base operation of the algebraic structure the query q outputs to
+    q: impl Fn(S) -> R, // The query over f that we want to check for linearity (to incrementalize)
```

Comment:
Can we move these comments into the doc comment ```/// ...`

Reply:
resolved in latest commit